### PR TITLE
Blog App. Milestone: Add Devise

### PR DESCRIPTION
**Pull request by CaptainBawa**
### On this branch:

- [ ] I completed the following:
     - Can register a new user.
     - User logs in with a combination of email and password.
     - Hashed passwords should be stored in the database.
     - Ask for confirmation by email.
     - Can reset password.

### CODE REVIEW

**Comment by nahnahdev** [comment]
## Project Status: INVALIDATED :red_circle: 
Hi @CaptainBawa and @Bellagirl-maker , 

While you made a great effort in this project, I can't go ahead and review your code.


## Invalid Code Review Request

- [ ] Great job so far Team. You did well with this project :clap: but currently, there is an error in your project that is preventing me from running all the migrations and starting your project to check if all the project requirements are met and if the functionality is met :+1: So kindly fix this  :wink: 

**- Error Message**
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by CaptainBawa** [comment]
### Hi @NahnahAJ 

I believe the issue might originate from your side, possibly due to a column in your database related to users, which is preventing my app from functioning correctly. On my end, everything works flawlessly. To confirm, please refer to the screenshot below, displaying the columns in my database.

Capture

**Comment by nahnahdev** [comment]
You can resubmit it for another review @CaptainBawa if you believe the issue is solved 👍

**Comment by CaptainBawa** [comment]
okay. Thank you

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
+
```
Comment (DeliceLydia) [inline app/models/user.rb]: - [ ] Please enable `:confirmable` in the user model so that the application asks for confirmation of email.

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @CaptainBawa, 

Good job so far! There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
-  I am able to sign_in and sign_up using `devise`
- No linter errors

## Required Changes ♻️
_Check the comments under the review._



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by cynthiainga — APPROVED** [review]
Hi @CaptainBawa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Users can signup and login to the app ✔️ 
- [ ] Descriptive PR ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**